packages/@react-aria/interactions/src/useLongPress.ts

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright 2020 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+
+// Portions of the code in this file are based on code from react.
+// Original licensing for the following can be found in the
+// NOTICE file in the root directory of this source tree.
+// See https://github.com/facebook/react/tree/cc7c1aece46a6b69b41958d731e0fd27c94bfc6c/packages/react-interactions
+
+import {PressEvent} from '@react-types/shared';
+import {usePress} from './usePress';
+import {useRef} from 'react';
+
+export interface LongPressHookProps {
+  onLongPress: (e: PressEvent) => void
+  triggerThreshold?: number
+}
+
+
+export function useLongPress(props : LongPressHookProps) {
+  let {
+    onLongPress,
+    triggerThreshold
+  } = props;
+  
+  triggerThreshold = triggerThreshold || 500;
+
+  const timeRef = useRef(null);
+
+  let {pressProps} = usePress({
+    onPressStart(e) {
+      timeRef.current = setTimeout(() => {
```

Comment:
I think we should only handle long press events for touch and mouse pointer types. Right now I noticed you can press and hold the enter or space keys on the keyboard as well. You could add a check here for `e.pointerType === 'mouse' || e.ponterType === 'touch'`.

Reply:
I agree with @devongovett, Enter and Space keys are tricky for a longPress on a menu trigger that opens a menu and sets focus to the first menuitem, because the keyup event might also trigger selection on the menuitem that received focus.